backend/internal/controllers/book.go

```diff
@@ -26,8 +27,21 @@ func (c *BookController) GetBookList(rw http.ResponseWriter, req *http.Request)
 func (c *BookController) GetBookDetailsByID(rw http.ResponseWriter, req *http.Request) {
 
 	var bookID string = chi.URLParam(req, "id")
+	var lang string = chi.URLParam(req, "lang")
+
+	valid, err := c.Books.CheckBookID(req.Context(), bookID, lang)
+	if err != nil {
+		writeResponse(rw, http.StatusInternalServerError, "error")
+		return
+	}
+
+	// Send 404 error if book does not exist
+	if !valid {
+		writeResponse(rw, http.StatusNotFound, "error")
```

Comment:
We can put descriptive error messages on 4xx requests, just not 500. Can this just say "book not found"?